### PR TITLE
miniserve: build on darwin

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/miniserve/default.nix [hunk 1]
```diff
@@ -1,28 +1,36 @@
...
   };
 
   cargoSha256 = "0ypxv9wqcnjxjdrvdparddpssrarnifr43dq7kcr4l3fd1anl40a";
 
   RUSTC_BOOTSTRAP = 1;
 
-  nativeBuildInputs = [ cmake pkg-config zlib ];
-  buildInputs = [ openssl ];
+  nativeBuildInputs = [ pkg-config zlib ];
+  buildInputs = stdenv.lib.optionals (!stdenv.isDarwin) [ openssl ]
```
Comment (veprbl) [inline pkgs/tools/misc/miniserve/default.nix:26]:
```suggestion
  buildInputs = if stdenv.isDarwin then [ Security ] else [ openssl ];
```

Comment (zowoq) [reply pkgs/tools/misc/miniserve/default.nix]: Thanks, updated.

**Comment by veprbl** [comment]
@GrahamcOfBorg build miniserve

**Review by veprbl — APPROVED** [review]
Builds on x86_64-darwin

**Pull request merged**